homeassistant/components/cloud/backup.py

```diff
@@ -146,9 +146,10 @@ async def async_upload_backup(
                 details["url"],
                 data=await open_stream(),
                 headers=details["headers"] | {"content-length": str(backup.size)},
+                timeout=ClientTimeout(connect=10.0, total=3600.0),
             )
             upload_status.raise_for_status()
-        except ClientError as err:
+        except (TimeoutError, ClientError) as err:
```

Comment:
Can we add a test?

Reply:
I take it it was not a question.
added.